internal/builtin/data_http.go

```diff
@@ -0,0 +1,271 @@
+	if r.Body != nil {
+		reqBody = strings.NewReader(*r.Body)
+	}
+
+	request, err := http.NewRequestWithContext(ctx, r.Method, u.String(), reqBody)
+	if err != nil {
+		return nil, err
+	}
+
+	if r.BasicAuthUsername != nil && r.BasicAuthPassword != nil {
+		request.SetBasicAuth(*r.BasicAuthUsername, *r.BasicAuthPassword)
+	}
+
+	if r.Headers != nil {
+		for k, v := range r.Headers {
+			request.Header.Set(k, v)
+		}
+	}
+
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: r.SkipVerify}
+	client := &http.Client{Transport: transport, Timeout: r.Timeout}
+
+	slog.Debug("Sending a HTTP request", "url", r.Url, "method", r.Method, "insecure", r.SkipVerify)
+
+	res, err := client.Do(request)
+	if err != nil {
+		return nil, err
+	}
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("the server responded with status code %d", res.StatusCode)
+	}
+
+	contentType := res.Header.Get("Content-Type")
+	var mimeType string
+	if contentType == "" {
+		mimeType = "text/plain" // assume `text/plain` if no content type set
+	} else {
+		mimeType, _, err = mime.ParseMediaType(contentType)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	defer res.Body.Close()
+
+	bytes, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error while reading response body: %s", err)
+	}
+
+	if !utf8.Valid(bytes) {
+		return nil, fmt.Errorf("response body is not recognized as UTF-8: %s", err)
+	}
+	return &Response{Body: bytes, MimeType: mimeType}, nil
+}
+
+func fetchHTTPDataWrapper(version string) plugin.RetrieveDataFunc {
+	return func(ctx context.Context, params *plugin.RetrieveDataParams) (plugin.Data, diagnostics.Diag) {
+		return fetchHTTPData(ctx, params, version)
+	}
+}
+
+func fetchHTTPData(ctx context.Context, params *plugin.RetrieveDataParams, version string) (plugin.Data, diagnostics.Diag) {
+	url := params.Args.GetAttr("url").AsString()
+	method := params.Args.GetAttr("method").AsString()
+	authUsername := params.Args.GetAttr("basic_auth_username")
+	authPassword := params.Args.GetAttr("basic_auth_username")
+	insecure := params.Args.GetAttr("insecure").True()
+	timeoutSecs, _ := params.Args.GetAttr("timeout_secs").AsBigFloat().Float64()
+	headers := params.Args.GetAttr("headers")
+	body := params.Args.GetAttr("body")
+
+	var req = Request{
+		Url:        url,
+		Method:     method,
+		Timeout:    time.Duration(timeoutSecs * float64(time.Second)),
+		SkipVerify: insecure,
+		Headers:    make(map[string]string),
+		Body:       nil,
+	}
+
+	if !authUsername.IsNull() && authUsername.AsString() != "" {
+		req.BasicAuthUsername = StringPtr(authUsername.AsString())
+	}
+
+	if !authPassword.IsNull() && authPassword.AsString() != "" {
+		req.BasicAuthPassword = StringPtr(authPassword.AsString())
+	}
+
+	if !headers.IsNull() {
+		for k, v := range headers.AsValueMap() {
+			req.Headers[k] = v.AsString()
+		}
+	}
+
+	if !body.IsNull() && body.AsString() != "" {
```

Comment:
`body` is ` cty.Map(cty.String)` type not `cty.String`. So `body.AsString()` will throw panic.

Reply:
my mistake, `body` must be a string